xwiki-platform-core/xwiki-platform-ratings/xwiki-platform-ratings-api/src/main/java/org/xwiki/ratings/script/RatingsScriptService.java

```diff
@@ -397,4 +411,42 @@ public AverageRatingApi getUserReputation(DocumentReference username)
             return null;
         }
     }
+
+    /**
+     * Retrieves the lowest level ratings configuration document in the hierarchy and fallback to XWiki.RatingsConfig
+     * if does not exist.
+     *
+     * @param documentRef the document being rated or for which the existing ratings are fetched
+     * @return the lowest level ratings configuration doc
+     * @since 8.1M1
+     */
+
+    public Document getConfigDoc(DocumentReference documentRef)
+    {
+        setError(null);
+
+        try {
+            SpaceReference lastSpaceReference = documentRef.getLastSpaceReference();
+            while (lastSpaceReference.getType() == EntityType.SPACE) {
+                DocumentReference spaceConfigDocReference =
+                    new DocumentReference(RatingsManager.RATINGS_CONFIG_SPACE_PAGE, lastSpaceReference);
+                XWikiDocument spaceConfigDoc = getXWiki().getDocument(spaceConfigDocReference, getXWikiContext());
+                if (spaceConfigDoc != null
+                    && spaceConfigDoc.getXObject(RatingsManager.RATINGS_CONFIG_CLASSREFERENCE) != null) {
+                    return spaceConfigDoc.newDocument(getXWikiContext());
+                }
+                if (lastSpaceReference.getParent().getType() == EntityType.SPACE) {
+                    lastSpaceReference = new SpaceReference(lastSpaceReference.getParent());
+                } else {
+                    break;
```

Comment:
Nope, it will break only when lastSpaceReference.getParent().getType() != EntityType.SPACE), but berofe that spaceConfigDoc.newDocument(getXWikiContext() will be returned if it's the case.